src/main/java/service/operation/PurchaseHandler.java

```diff
@@ -0,0 +1,13 @@
+package service.operation;
+
+import db.Storage;
+import models.FruitTransaction;
+
+public class PurchaseHandler implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction fruitTransaction) {
+        Integer oldQuantity = Storage.fruitMap.get(fruitTransaction.getFruit());

```

Comment:
what will happen if `fruitMap.get()` returns `null`? you can use `fruitMap.getOrDefault()` instead.